pkg/controller/vsphere/machine_scope.go

```diff
@@ -130,6 +130,44 @@ func (s *machineScope) GetSession() *session.Session {
 	return s.session
 }
 
+func (s *machineScope) isNodeLinked() bool {
+	return s.machine.Status.NodeRef != nil && s.machine.Status.NodeRef.Name != ""
```

Comment:
Maybe this can be transferred to machine structure function and used instead https://github.com/openshift/machine-api-operator/blob/f4110d923215831c8d47d846f82b5d8ce156c2e8/pkg/controller/machine/controller.go#L518?

The same might apply to getNode(), we might need this in the future.

Reply:
I'm for it, but think follow-up with wider refactoring would be better there.